docs/DesignDocs/RemoteAttestationCollaterals.md

```diff
@@ -0,0 +1,488 @@
+    size_t* evidence_buffer_size,
+    uint8_t** endorsements_buffer,
+    size_t* endorsements_buffer_size);
+
+/**
+ * Free up any resources allocated by oe_get_evidence()
+ *
+ * @param[in] evidence_buffer THe buffer containing the evidence.
+ * @param[in] endorsements_buffer The buffer containing the endorsements.
+ */
+void oe_free_evidence(
+    uint8_t* endorsements_buffer,
+    uint8_t* evidence_buffer);
+
+/**
+ * Verify the integrity of the evidence and its signature,
+ * with endorsements that are associated with the evidence.
+ * This function works for both local and remote attestation.
+ *
+ * This function is available in the enclave as well as in the host.
+ *
+ * @param[in] evidence_buffer The buffer containing the evidence to verify.
+ * @param[in] evidence_buffer_size The size of the **evidence** buffer.
+ * @param[in] endorsements Optional The endorsement data that is associated with
+ * the evidence.
+ * @param[in] endorsements_size The size of the **endorsements** buffer.
+ * @param[in] input_validation_time Optional datetime to use when verifying
+ * evidence. If not specified, it will use the creation_datetime of the
+ * endorsements (if any endorsements are provided).
+ * @param[out] claims The list of claims.
+ * @param[out] claims_size The size of claims.
+ *
+ * @retval OE_OK The verification was successful.
+ * @retval OE_INVALID_PARAMETER At least one parameter is invalid.
+ *
+ */
+oe_result_t oe_verify_evidence(
+    const uint8_t* evidence_buffer,
+    size_t evidence_buffer_size,
+    const uint8_t* endorsements,
+    size_t endorsements_size,
+    oe_datetime_t* input_validation_time,
+    uint8_t** claims,
+    size_t* claims_size);
+```
+
+### Claims
+As part of the claims form `oe_verify_evidence()`, there will be a validity
+datetime range, `validity_from` and `validity_until` claims that applies to the evidence and endorsements.
+
+Current set of claims definitions:
+
+| Claim Name       | Claim Value Type   | Description                                                          |
+|:-----------------|:-------------------|:---------------------------------------------------------------------|
+| id_version       | uint32_t           | Claims version. Must be 0                                            |
+| security_version | uint32_t           | Security version of the enclave. (ISVN for SGX).                     |
+| attributes       | uint64_t           | Attributes flags for the evidence: <br/> `OE_REPORT_ATTRIBUTES_DEBUG`: The evidence is for a debug enclave.<br/> `OE_REPORT_ATTRIBUTES_REMOTE`: The evidence can be used for remote attestation.   |
+| unique_id        | uint8_t[32]        | The unique ID for the enclave (MRENCLAVE for SGX).                   |
+| signer_id        | uint8_t[32]        | The signer ID for the enclave (MRSIGNER for SGX).                    |
+| product_id       | uint8_t[32]        | The product ID for the enclave (ISVPRODID for SGX).                  |
```

Comment:
Please have a look at https://github.com/openenclave/openenclave/pull/2262#discussion_r337815502 and at https://github.com/openenclave/openenclave/pull/2262#discussion_r337815883 and let me know what you think.

Reply:
I responded in the #2262 PR thread.